pkg/proxy/iptables/proxier.go

```diff
@@ -1373,29 +1384,67 @@ func (proxier *Proxier) syncProxyRules() {
 			proxier.writeServiceToEndpointRules(svcNameString, svcInfo, clusterPolicyChain, clusterEndpoints, args)
 		}
 
-		if svcInfo.UsesLocalEndpoints() {
```

Comment:
again, this didn't cherry-pick correctly and you'll need to figure out where to put the code in the 1.24 version